src/mscorlib/src/System/Globalization/CompareInfo.cs

```diff
@@ -459,6 +491,34 @@ internal static unsafe int CompareOrdinalIgnoreCase(string strA, int indexA, int
                     length--;
                 }
 
+                if (CultureData.InvariantMode)
```

Comment:
Or a separate function as @stephentoub suggests. It's just confusing two have two blocks like this IMO.